### PR TITLE
Fixed #27557 -- Casted GEOSGeometry only when necessary

### PR DESCRIPTION
**Pull request by claudep**
Thanks Pete Flugstad for the report.

### CODE REVIEW

File: tests/gis_tests/geos_tests/test_geos.py [hunk 1]
```diff
@@ -1308,6 +1308,20 @@ def test_deconstructible(self):
         self.assertEqual(args, (Point(0, 0), MultiPoint(Point(0, 0), Point(1, 1)), poly))
         self.assertEqual(kwargs, {})
 
+    def test_subclassing(self):
+        class ExtendedPolygon(Polygon):
```
Comment (timgraham) [inline tests/gis_tests/geos_tests/test_geos.py]: Possibly a docstring could be useful?  `"""GEOSGeometry may be subclassed ....""""`

File: tests/gis_tests/geos_tests/test_geos.py [hunk 2]
```diff
@@ -1308,6 +1308,20 @@ def test_deconstructible(self):
...
+                data = kwargs.pop('data', 0)
+                super(ExtendedPolygon, self).__init__(*args, **kwargs)
+                self._data = data
+
+            def __str__(self):
+                return "EXT_POLYGON - data: %d - %s" % (self._data, self.wkt)
+
+        ext_poly = ExtendedPolygon(((0, 0), (0, 1), (1, 1), (0, 0)), data=3)
+        self.assertEqual(type(ext_poly), ExtendedPolygon)
+        self.assertEqual(str(ext_poly), "EXT_POLYGON - data: 3 - POLYGON ((0 0, 0 1, 1 1, 0 0))")
```
Comment (timgraham) [inline tests/gis_tests/geos_tests/test_geos.py]: Is it unneeded complexity to verify `ext_poly._data` via `__str__()` as opposed to checking the attribute directory?

Comment (claudep) [inline tests/gis_tests/geos_tests/test_geos.py]: We are not testing the attribute presence (or at least not only), but mainly that the good `__str__` method is called (the overriden `__str__` method was skipped before).

Comment (timgraham) [inline tests/gis_tests/geos_tests/test_geos.py]: Explaining that in the docstring would be great.

**Comment by claudep** [comment]
What about the added comments?

**Comment by timgraham** [comment]
Makes sense.

**Pull request closed without merging**